src/OpenTelemetry/Internal/SelfDiagnosticsEventListener.cs

```diff
@@ -143,10 +145,10 @@ internal void WriteEvent(string eventMessage, ReadOnlyCollection<object> payload
                 // Not using foreach because it can cause allocations
                 for (int i = 0; i < payload.Count; ++i)
                 {
-                    object obj = payload[i];
+                    object? obj = payload[i];
                     if (obj != null)
                     {
-                        pos = EncodeInBuffer(obj.ToString(), true, buffer, pos);
+                        pos = EncodeInBuffer(obj.ToString() ?? "null", true, buffer, pos);
```

Comment:
Could `obj.ToString()` return `null` for any of the payloads that our `EventSource` events log?

Reply:
The contract for `ToString()` is:

```csharp
    public partial class Object
    {
        public virtual string? ToString()
        {
            return GetType().ToString();
        }
    }
```

So if someone _really_ wanted to, it is perfectly valid to do this:

```csharp
public class MyClass
{
    public override string? ToString() => null;
}
```

Do any of the existing logs do that? Doubtful, but I don't think it is a bad idea to be defensive here.

Reply:
Maybe `obj.ToString() ?? object.GetType().ToString()` would be better than `obj.ToString() ?? "null"` here?

Reply:
Well, if someone really wanted to, they could also throw an exception in the `ToString()` implementation. We don't bother checking for that though. 😀

I guess it's probably okay to be defensive here since the nullable decorations complain about it.

I would prefer having  `obj.ToString() ?? "null"` over `obj.ToString() ?? object.GetType().ToString()` as it would not help when the events log value-types as payload?

Reply:
Fine by me to leave it "null" if you like it that way! We could just do `obj.ToString()!` to quiet the warning but I feel better about doing `obj.ToString() ?? "null"` because runtime is explicitly saying `null` is OK via the annotation.